ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java

```diff
@@ -259,13 +258,14 @@ public PendingTransactions getPendingTransactions() {
               "Transaction gas limit of %s exceeds block gas limit of %s",
               transaction.getGasLimit(), chainHeadBlockHeader.getGasLimit()));
     }
-    if (transaction.getType().equals(TransactionType.EIP1559)
-        && !eip1559
-            .map(eip1559 -> eip1559.isEIP1559(chainHeadBlockHeader.getNumber()))
-            .orElse(false)) {
-      return ValidationResult.invalid(
-          TransactionInvalidReason.INVALID_TRANSACTION_FORMAT,
-          "EIP-1559 transaction are not allowed yet");
+    if (transaction.getType().equals(TransactionType.EIP1559)) {
+      final long blknum = chainHeadBlockHeader.getNumber();
+      ProtocolSpec spec = protocolSchedule.getByBlockNumber(blknum);
+      if (!spec.getEip1559().map(eip1559 -> eip1559.isEIP1559(blknum)).orElse(false)) {
```

Comment:
This is an improvement on readability. Could we take it a step further and push all the navigation of the Optional into the Protocol spec itself?

Reply:
might come in handy below as well.

Reply:
I am going to defer this until the subsequent cleanup PR, because I am intending to remove the Optional<EIP1559> entirely.